assets/src/edit-story/components/library/panes/text/textSets/getTextSets.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { migrate } from '../../../../../migration/migrate';
+
+function updateMinMax(minMax, element) {
+  // Purposely mutating object so passed
+  // in minMax is modified
+  minMax.minX = Math.min(minMax.minX, element.x);
+  minMax.maxX = Math.max(minMax.maxX, element.x + element.width);
+
+  minMax.minY = Math.min(minMax.minY, element.y);
+  minMax.maxY = Math.max(minMax.maxY, element.y + element.height);
+
+  return minMax;
+}
+
+async function loadTextSet(name) {
+  const data = await import(`./raw/${name}.json`);
+  const migrated = migrate(data, data.version);
+
+  const textSets = migrated.pages.reduce((sets, page) => {
+    const minMax = {
+      minX: Infinity,
+      maxX: 0,
+      minY: Infinity,
+      maxY: 0,
+    };
+
+    const textElements = page.elements.filter((element) => {
+      return !element.isBackground && Boolean(updateMinMax(minMax, element));
+    });
+
+    return [
+      ...sets,
+      textElements.map((e) => ({
+        ...e,
+        // Offset elements so the text set's
+        // default position is (0,0)
+        x: e.x - minMax.minX,
+        y: e.y - minMax.minY,
```

Comment:
This shouldn't be done here - only when previewing the elements in the text set panel. When actually inserted, the "original" placement of the elements should be used. Right now they're just added to the top left corner regardless of the actual position on the page when designed.